ProjFS.Linux/PrjFSLib.Linux.Managed/Interop/ProjFS.cs

```diff
@@ -131,5 +210,13 @@ public struct Handlers
             public EventHandler HandleNotifyEvent;
             public EventHandler HandlePermEvent;
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct Attr
+        {
+            public IntPtr Name;
+            public IntPtr Value;
```

Comment:
I'd be inclined to try defining these as `byte*`, for additional type safety.

Reply:
OK -- I figured out how to do this, which involves marking the `struct` as `unsafe`, but before I commit, may I ask if this kind of change should also be applied to the `struct Event` as well?  It has a couple of pointers-to-strings (the `Path` and `TargetPath` members).

Would it make sense to convert those as well?  Or not, because they are only read by the C# code?  I suppose if we did convert them to `byte*`, the `PtrToStringUTF8()` method in `VirtualizationInstance.cs` would no longer need to cast the `IntPtr`s to `byte*`, at least:
```
return Encoding.UTF8.GetString((byte*)ptr, length);
```
but it would presumably need to be marked as an `unsafe` method, as would a number of others in that class.  Advice?

Reply:
It's all a matter of taste, which afaict can only really be evaluated here by trying out the change and deciding whether it's uglier or less ugly as a result. If the `unsafe` calls ripple out too far, perhaps not. Ideally we would be able to fully encapsulate the ugliness in safe functions which unsafely pick apart the struct.

Reply:
I think the ripples will be fairly extensive, on a first try.

Reply:
Actually, that turned out not too bad, in my estimation.  See what you think of b57c222, @kivikakk.